<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>View</class>
 <widget class="QMainWindow" name="View">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>520</width>
    <height>722</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>550</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>View</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>-10</x>
      <y>0</y>
      <width>551</width>
      <height>671</height>
     </rect>
    </property>
    <property name="tabPosition">
     <enum>QTabWidget::North</enum>
    </property>
    <property name="tabShape">
     <enum>QTabWidget::Rounded</enum>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <property name="elideMode">
     <enum>Qt::ElideRight</enum>
    </property>
    <property name="usesScrollButtons">
     <bool>false</bool>
    </property>
    <property name="documentMode">
     <bool>true</bool>
    </property>
    <property name="tabsClosable">
     <bool>false</bool>
    </property>
    <widget class="QWidget" name="Maze">
     <attribute name="title">
      <string>Maze</string>
     </attribute>
     <widget class="QPushButton" name="pushButton">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>90</y>
        <width>98</width>
        <height>32</height>
       </rect>
      </property>
      <property name="focusPolicy">
       <enum>Qt::NoFocus</enum>
      </property>
      <property name="text">
       <string>Solve Maze</string>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <property name="checked">
       <bool>false</bool>
      </property>
      <property name="autoDefault">
       <bool>false</bool>
      </property>
      <property name="default">
       <bool>false</bool>
      </property>
      <property name="flat">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QSpinBox" name="sbAY">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>20</y>
        <width>42</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="sbAX">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>0</y>
        <width>42</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
     </widget>
     <widget class="MazeWidget" name="widget" native="true">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>500</width>
        <height>500</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>300</width>
        <height>300</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>500</width>
        <height>500</height>
       </size>
      </property>
     </widget>
     <widget class="QSpinBox" name="sbBX">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>50</y>
        <width>42</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
     </widget>
     <widget class="QLabel" name="labelBY">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>19</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>B y</string>
      </property>
     </widget>
     <widget class="Line" name="line">
      <property name="geometry">
       <rect>
        <x>-10</x>
        <y>110</y>
        <width>651</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="labelAX">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>19</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>A x</string>
      </property>
     </widget>
     <widget class="QLabel" name="labelBX">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>19</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>B x</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="sbBY">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>70</y>
        <width>42</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
     </widget>
     <widget class="QLabel" name="labelAY">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>19</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>A y</string>
      </property>
     </widget>
     <widget class="QPushButton" name="generateButton">
      <property name="geometry">
       <rect>
        <x>270</x>
        <y>40</y>
        <width>100</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Generate</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="sbGenW">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>30</y>
        <width>51</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>3</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="sbGenH">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>60</y>
        <width>51</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>3</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
     </widget>
     <widget class="QLabel" name="labelWidth">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>30</y>
        <width>36</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Width</string>
      </property>
     </widget>
     <widget class="QLabel" name="labelHeight">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>60</y>
        <width>40</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Height</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="Cave">
     <attribute name="title">
      <string>Cave</string>
     </attribute>
     <widget class="Line" name="line2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>110</y>
        <width>651</width>
        <height>16</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="labelDelay">
      <property name="geometry">
       <rect>
        <x>379</x>
        <y>90</y>
        <width>65</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Delay (ms)</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="sbDelay">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>90</y>
        <width>61</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>10</number>
      </property>
      <property name="maximum">
       <number>10000</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
     </widget>
     <widget class="CaveWidget" name="widgetCave" native="true">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>130</y>
        <width>500</width>
        <height>500</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="genStartStop">
      <property name="geometry">
       <rect>
        <x>315</x>
        <y>85</y>
        <width>59</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Start</string>
      </property>
     </widget>
     <widget class="QPushButton" name="genCaveButton">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>80</y>
        <width>100</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Generate</string>
      </property>
     </widget>
     <widget class="QLabel" name="labelHeight2">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>50</y>
        <width>40</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Height</string>
      </property>
     </widget>
     <widget class="QLabel" name="labelWidth2">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>20</y>
        <width>36</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Width</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="sbGenWCave">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>20</y>
        <width>51</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>4</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>20</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="sbGenHCave">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>50</y>
        <width>51</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>4</number>
      </property>
      <property name="maximum">
       <number>50</number>
      </property>
      <property name="value">
       <number>20</number>
      </property>
     </widget>
     <widget class="QPushButton" name="genStep">
      <property name="geometry">
       <rect>
        <x>230</x>
        <y>86</y>
        <width>59</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>Step</string>
      </property>
     </widget>
     <widget class="QLabel" name="labelInitChnc">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>30</y>
        <width>118</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Initialization chance</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="sbInit">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>50</y>
        <width>51</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>0</number>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>50</number>
      </property>
     </widget>
     <widget class="QLabel" name="labelBirthLim">
      <property name="geometry">
       <rect>
        <x>389</x>
        <y>23</y>
        <width>58</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Birth limit</string>
      </property>
     </widget>
     <widget class="QLabel" name="labelDeathLim">
      <property name="geometry">
       <rect>
        <x>382</x>
        <y>53</y>
        <width>65</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Death limit</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="sbBLim">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>20</y>
        <width>42</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>7</number>
      </property>
     </widget>
     <widget class="QSpinBox" name="sbDLim">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>50</y>
        <width>42</width>
        <height>22</height>
       </rect>
      </property>
      <property name="maximum">
       <number>7</number>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>520</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MazeWidget</class>
   <extends>QWidget</extends>
   <header location="global">./headers/mazewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>CaveWidget</class>
   <extends>QWidget</extends>
   <header>./headers/cavewidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>sbAX</sender>
   <signal>valueChanged(int)</signal>
   <receiver>pushButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>40</x>
     <y>56</y>
    </hint>
    <hint type="destinationlabel">
     <x>38</x>
     <y>151</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbAY</sender>
   <signal>valueChanged(int)</signal>
   <receiver>pushButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>40</x>
     <y>76</y>
    </hint>
    <hint type="destinationlabel">
     <x>38</x>
     <y>151</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbBX</sender>
   <signal>valueChanged(int)</signal>
   <receiver>pushButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>40</x>
     <y>106</y>
    </hint>
    <hint type="destinationlabel">
     <x>38</x>
     <y>151</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbBY</sender>
   <signal>valueChanged(int)</signal>
   <receiver>pushButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>40</x>
     <y>126</y>
    </hint>
    <hint type="destinationlabel">
     <x>38</x>
     <y>151</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
